### PR TITLE
"Donor Profiles" are now renamed "Donor Dashboard"

### LINKED ISSUE #5682: "Donor Profiles" should be renamed "Donor Dashboard"
## User Story

As a donor, I want Donor Profiles to be renamed Donor Dashboard so that I have more clear expectations for the feature.

## Details

From Canny;
"Donor Dashboard" is a more clear description of the new functionality. We as a product management team have decided to change all places where the words "Donor Profile" are currently to "Donor Dashboard." This includes the naming of the shortcode, block, and all settings and notices in the admin.

## Additional Context

This PR should not update the domain namespace of DonorProfiles. That will be saved for a future PR, as it affects multiple different addons.

## Acceptance Criteria

-   [x] Written instances of Donor Profiles are updated to use Donor Dashboard
-   [x] Default slug is updated from "donor-profile" to "donor-dashboard"

### PR DESCRIPTION
**Pull request by henryholtgeerts**
Resolves #5682 

## Description

This PR renames references to Donor Profiles as Donor Dashboards. This PR should change most references in the codebase, and all visual references to Donor Dashboards. Notably: The PR does not update the domain namespace, which is still DonorProfiles. This will be addressed in a future PR, when addons which rely on the domain namespace will also be updated.

## Affects

This PR affects many, many facets of the DonorProfiles domain. Primarily, it is meant to ensure that any visual representation of the donor profile feature is now referred to as the Donor Dashboard.

## Visuals

N/A

## Testing Instructions

1. Setup a new Donor Dashboard
2. Test various functionality to ensure that nothing is newly broken
3. Check copy for ferences to Donor Profiles (including contextual docs, generated page title, block name and shortcode).

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/DonorProfiles/resources/js/app/components/rest-subscription-table/index.js [hunk 1]
```diff
@@ -8,16 +8,16 @@ const RESTSubscriptionTable = () => {
 		<Table
 			header={
 				<Fragment>
-					<div className="give-donor-profile-table__column">
+					<div className="give-donor-dashboard-table__column">
 						Subscription
```
Comment (ravinderk) [inline src/DonorProfiles/resources/js/app/components/rest-subscription-table/index.js:12]: @henryholtgeerts Can we localize these strings?

File: src/DonorProfiles/resources/js/app/components/auth-modal/index.js [hunk 2]
```diff
@@ -72,60 +72,60 @@ const AuthModal = () => {
...
+				<div className="give-donor-dashboard__auth-modal-heading">
 					{ __( 'Log in to see your donor dashboard', 'give' ) }
 				</div>
-				<div className="give-donor-profile__auth-modal-content">
+				<div className="give-donor-dashboard__auth-modal-content">
 					{ loggedInWithoutDonor && (
-						<div className="give-donor-profile__auth-modal-notice">
-							{ __( 'The account you are currently logged into the site with does not have an associated donor profile. Donate now or contact the site administrator to associate this WordPress user with a donor profile.' ) }
+						<div className="give-donor-dashboard__auth-modal-notice">
+							{ __( 'The account you are currently logged into the site with does not have an associated donor dashboard. Donate now or contact the site administrator to associate this WordPress user with a donor dashboard.' ) }
```
Comment (ravinderk) [inline src/DonorProfiles/resources/js/app/components/auth-modal/index.js:83]: Can we add the missing text domain?

File: src/DonorProfiles/resources/js/app/components/rest-subscription-table/subscription-row/index.js [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
+		<div className="give-donor-dashboard-table__row">
+			<div className="give-donor-dashboard-table__column">
+				<div className="give-donor-dashboard-table__subscription-amount">$200 / Monthly</div>
+				<div className="give-donor-dashboard-table__subscription-campaign">Save Friar Tuck</div>
 			</div>
-			<div className="give-donor-profile-table__column">
-				<div className="give-donor-profile-table__subscription-status">
+			<div className="give-donor-dashboard-table__column">
+				<div className="give-donor-dashboard-table__subscription-status">
 					Active
```
Comment (ravinderk) [inline src/DonorProfiles/resources/js/app/components/rest-subscription-table/subscription-row/index.js:10]: Can we localize these strings?

File: src/DonorProfiles/resources/views/successnotice.php [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
-$pageId = give_get_option( 'donor_profile_page' );
+$pageId = give_get_option( 'donor_dashboard_page' );
 
 $pageUrl = get_permalink( $pageId );
 
 ?>
 
 <div class="notice notice-success is-dismissible">
-	<p><?php printf( __( 'Success! Donor Profile page was created. You can <a href="%s">take a look at it here.</a>' ), $pageUrl ); ?></p>
+	<p><?php printf( __( 'Success! Donor Dashboard page was created. You can <a href="%s">take a look at it here.</a>' ), $pageUrl ); ?></p>
```
Comment (ravinderk) [inline src/DonorProfiles/resources/views/successnotice.php:10]: Can we localize this string?

**Review by ravinderk — CHANGES_REQUESTED** [review]
@henryholtgeerts I find that few strings are not localized. I was not sure about them. Can you take a look?

File: src/DonorProfiles/resources/js/app/components/rest-subscription-table/index.js (hunk 1 above)
Comment (henryholtgeerts) [reply src/DonorProfiles/resources/js/app/components/rest-subscription-table/index.js]: Ah good catch! This components is actually meant to be removed (replaced by SubscriptionTable instead). I've now removed this logic, ref: https://github.com/impress-org/givewp/pull/5708/commits/ba86799ecdac13b18d894dffc003e970b25b23d5

File: src/DonorProfiles/resources/js/app/components/rest-subscription-table/subscription-row/index.js (hunk 3 above)
Comment (henryholtgeerts) [reply src/DonorProfiles/resources/js/app/components/rest-subscription-table/subscription-row/index.js]: This component is unused and has been removed, ref: https://github.com/impress-org/givewp/pull/5708/commits/ba86799ecdac13b18d894dffc003e970b25b23d5

File: src/DonorProfiles/resources/js/app/components/auth-modal/index.js (hunk 2 above)
Comment (henryholtgeerts) [reply src/DonorProfiles/resources/js/app/components/auth-modal/index.js]: Done, ref: https://github.com/impress-org/givewp/pull/5708/commits/4fb4d2daea978aa6b2006a3ea5459febfd146784

File: src/DonorProfiles/resources/views/successnotice.php (hunk 4 above)
Comment (henryholtgeerts) [reply src/DonorProfiles/resources/views/successnotice.php]: Done, ref: https://github.com/impress-org/givewp/pull/5708/commits/d68a55292442e50d45fb4be16dfc7b0cf25baf26

**Comment by henryholtgeerts** [comment]
@ravinderk Initial feedback about localization has all been incorporated and addressed! Thanks, look forward to review re: functionality.

**Review by ravinderk — APPROVED** [review]
@henryholtgeerts code looks good to me and the Initial test shows that existing functionality does not break.

**Pull request merged**